source/core/Variants.cpp

```diff
@@ -439,6 +439,11 @@ bool VariantsAreEqual(VariantTypeRef variantX, VariantTypeRef variantY)
 {
     DualTypeVisitor visitor;
     DualTypeEqual dualTypeEqual;
+    if (!variantX && !variantY) {
```

Comment:
I think we need to move this check inside the visitor or the operation. For inner variants this case may present again.